frontend/src/PlayerRemapForm.tsx

```diff
@@ -0,0 +1,179 @@
+        { errorOnSubmit, successMessage, loading },
+        { handleInputChange, validateField, handleSubmit },
+    ] = useFormData<PlayerRemapFormData, ValidPlayerRemapFormData>({
+        initialFormData,
+        optionalFields: [],
+        missingFieldErrorMessage: ErrorMessage.ExistingPlayerRequired,
+        submit,
+        toErrorMessage,
+    });
+
+    const [warningAlert, setWarningAlert] = useState<string | null>(null);
+
+    useEffect(
+        function refreshOnSubmit() {
+            if (successMessage) {
+                refresh();
+            }
+        },
+        [successMessage]
+    );
+
+    return (
+        <Sheet
+            sx={{
+                display: "flex",
+                flexDirection: "column",
+                alignItems: "center",
+                justifyContent: "center",
+            }}
+        >
+            {successMessage ? (
+                <Typography color="success">{successMessage}</Typography>
+            ) : (
+                <>
+                    <Typography level="title-lg">{name}</Typography>
+
+                    <Box sx={{ my: 2, width: "100%" }}>
+                        <FormControl error={!!formData.toPlayer.error}>
+                            <FormLabel>
+                                Reassign this player's games to:
+                            </FormLabel>
+
+                            <Autocomplete
+                                current={formData.toPlayer.value}
+                                options={playerOptions}
+                                placeholder="Player name"
+                                loading={false}
+                                onChange={handleInputChange("toPlayer")}
+                                validate={validateField("toPlayer")}
+                            />
+
+                            {formData.toPlayer.error && (
+                                <FormHelperText>
+                                    {formData.toPlayer.error}
+                                </FormHelperText>
+                            )}
+                        </FormControl>
+
+                        {warningAlert && (
+                            <Alert color="warning" sx={{ my: "10px" }}>
+                                {warningAlert}
+                            </Alert>
+                        )}
+                    </Box>
+
+                    <Button
+                        onClick={() => {
+                            if (warningAlert || formData.toPlayer.error) {
+                                handleSubmit();
+                            } else {
+                                setWarningAlert(
+                                    `Danger: ${name} will cease to exist. ${formData.toPlayer.value?.label} will absorb all of ${name}'s history. ${name}'s history cannot be recovered. If you're sure, press "Submit" again to continue.`
+                                );
+                            }
+                        }}
```

Comment:
This is the fake-out alert step. There's a small UI bug I didn't have the energy to work out. It happens if you follow these steps:

1. Submit the form, read "danger" alert, submit again. Receive an error from the server. The error is displayed (expected)

2. Submit again. "Danger" alert is displayed (expected). But, the old submission error from the server is not cleared

This is because errors are cleared by the `handleSubmit` provided by `useFormData`, which isn't called until after the alert step. I think the solution is pass the alert message into `useFormData` so `useFormData` can handle it in the proper order. Unless you think it's urgent, will make an issue and handle it in a follow-up, maybe post-MVP.

Reply:
I don't think it's urgent! This looks good to me.